src/queries/a11yStates.ts

```diff
@@ -0,0 +1,79 @@
+import type { ReactTestInstance } from 'react-test-renderer';
+import { AccessibilityState } from 'react-native';
+import { makeQueries } from './makeQueries';
+import type {
+  FindAllByQuery,
+  FindByQuery,
+  GetAllByQuery,
+  GetByQuery,
+  QueryAllByQuery,
+  QueryByQuery,
+} from './makeQueries';
+
+type AccessibilityStateKey = keyof AccessibilityState;
+type AccessibilityStateKeyQueryParam =
+  | AccessibilityStateKey
+  | Array<AccessibilityStateKey>;
+
+export function matchArrayValue(
+  prop: Array<string> | undefined,
+  matcher: string | Array<string>
+): boolean {
+  if (!prop || matcher.length === 0) {
+    return false;
+  }
+
+  if (typeof matcher === 'string') {
+    return prop.includes(matcher);
+  }
+
+  return !matcher.some((e) => !prop.includes(e));
```

Comment:
I don't know what it is the desired behaviour, but I would expect to require all the states

Reply:
Indeed I don't know, though I did not change the logic, I copy pasted the `matchArrayValue` function from the previous a11yApi file. Maybe we should open an issue to discuss it? (as this PR should not change any behavior)

Reply:
Hmm I think it's ok in the end, because it checks that all the values in the matcher are included in the prop accessiblityStates, not the other way around
so : 
```js
matchArrayValue(["selected", "disabled"], [”selected"]) // returns true
matchArrayValue(["selected"], [”selected", "disabled"]) // returns false
```
You can check the tests I did [here](https://github.com/callstack/react-native-testing-library/pull/977/commits/24f5259ab85d7fb1514a9127a4725faac4a17978)

I tried to make the function more explicit in this [commit](https://github.com/callstack/react-native-testing-library/pull/977/commits/5c92cb5e9592fb7b2c3ee4e726917011871194c9)